src/app/components/shared/datetime/datetime/datetime.component.ts

```diff
@@ -0,0 +1,132 @@
+  selector: "baw-datetime",
+  templateUrl: "../abstract-datetime.component.html",
+})
+export class DatetimeComponent extends AbstractDatetimeComponent {
+  public constructor() {
+    super();
+  }
+
+  @Input({ required: true }) public value!: DateTime | Date;
+  @Input() public dateOnly: string | boolean;
+  @Input() public timeOnly: string | boolean;
+  protected timezone?: BawTimezoneUnion;
+
+  // we create a getter for the a luxon DateTime object so that we can get values from the @Input() value
+  // without type guards, shape, and instance checks
+  protected get luxonDateTime(): DateTime {
+    const luxonDate: DateTime =
+      this.value instanceof Date
+        ? DateTime.fromJSDate(this.value, { zone: "utc" })
+        : this.value;
+
+    // localize the date/time to the users timezone using toLocal()
+    return luxonDate.toLocal();
+  }
+
+  public update(): void {
+    // the fullDateTime is used regardless of format as it is used in the tooltip
+    // it emits as much information as possible
+    const fullDateTime = this.luxonDateTime.toFormat("yyyy-MM-dd HH:mm:ss.SSS");
+    const timezoneName = this.longTimezone();
+
+    // we do not place the timezoneName in brackets as the ngx-bootstrap tooltip
+    // is an "absolute" width. This means that the brackets will be split over multiple lines
+    // degrading the readability of the tooltip and user experience
+    this.tooltipValue = `${fullDateTime} ${timezoneName}`;
+    this.rawDateTime = this.luxonDateTime.toISO();
+    this.formattedValue = this.luxonDateTime.toFormat(this.dateTimeFormat());
+  }
+
+  protected dateTimeFormat(): string {
+    //! we should prefer not to use this functionality
+    // however, by allowing the programmer to specify both dateOnly and timeOnly to be true
+    // we can do reactive formats such as <baw-datetime [dateOnly]="isDateOnly" [timeOnly]="isTimeOnly">
+    // it also exhaustively fills all possible combinations of dateOnly and timeOnly, reducing the obfuscation of the code
+    // eg. without this condition, setting both dateOnly and timeOnly to true would result in the dateOnly format being used
+    // which is not obvious to the programmer without looking at the code
+    if (
+      isInstantiated(this.dateOnly) &&
+      this.dateOnly !== false &&
+      isInstantiated(this.timeOnly) &&
+      this.timeOnly !== false
+    ) {
+      return "yyyy-MM-dd HH:mm:ss";
+    }
+
+    // we use isInstantiated() to check if the humanized attribute is set
+    // and we also check that it is not set to false
+    // by allowing boolean values, we support reactive formats (e.g. [dateOnly]="isDateOnly")
+    if (isInstantiated(this.dateOnly) && this.dateOnly !== false) {
+      return "yyyy-MM-dd";
+    } else if (isInstantiated(this.timeOnly) && this.timeOnly !== false) {
+      return "HH:mm:ss";
+    }
+
+    // include both date and time if neither dateOnly or timeOnly are specified
+    return "yyyy-MM-dd HH:mm:ss";
+  }
+
```

Comment:
why "we should prefer not to use this functionality"?

also:
- constantize the strings
- check values only once
- simplify the branches.

```
const showDate = !(this.dateOnly === "" || this.dateOnly === true)
const showtime = !(this.timeOnly === "" || this.timeOnly === true)

if (showDate && showTime ) {
    return FULL_DATE;
}
else if (showDate) {
    return DATE_ONLY;
}
else {
    return TIME_ONLY;
}
```

and if we take advantage of the new `transform` option for `@Input` we can do even better: https://stackoverflow.com/a/76537839/224512

Reply:
and again, I haven't looked yet but this seems like functionality that belongs in the abstract class

Reply:
In retrospect, this could also be implemented as a simple Fizz Buzz method